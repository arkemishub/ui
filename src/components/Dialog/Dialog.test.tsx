/**
 * Copyright 2023 Arkemis S.r.l.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

import { render } from "@testing-library/react";
import { Dialog } from "./index";
import { userEvent } from "@storybook/testing-library";

const intersectionObserverMock = () => ({
  observe: () => null,
  disconnect: () => null,
});
window.IntersectionObserver = jest
  .fn()
  .mockImplementation(intersectionObserverMock);

describe("Dialog", () => {
  test("it should match snapshot", () => {
    const { asFragment } = render(<Dialog open={true} onClose={() => null} />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("should render dialog", () => {
    const { getByTestId } = render(<Dialog open={true} onClose={() => null} />);
    expect(getByTestId("arke-dialog")).toBeInTheDocument();
  });

  it("should not render dialog when open prop is false", () => {
    const { queryAllByTestId } = render(
      <Dialog open={false} onClose={() => null} />
    );
    expect(queryAllByTestId("arke-dialog")).toHaveLength(0);
  });

  it("should call onClose when close button is clicked", () => {
    const onClose = jest.fn();
    const { getByTestId } = render(<Dialog open={true} onClose={onClose} />);
    userEvent.click(getByTestId("arke-dialog-close"));
    expect(onClose).toHaveBeenCalled();
  });

  it("should render title", () => {
    const { getByText } = render(
      <Dialog open={true} onClose={() => null} title="Title" />
    );
    expect(getByText("Title")).toBeInTheDocument();
  });

  it("should render children", () => {
    const { getByText } = render(
      <Dialog open={true} onClose={() => null} title="Title">
        Children
      </Dialog>
    );
    expect(getByText("Children")).toBeInTheDocument();
  });
});
