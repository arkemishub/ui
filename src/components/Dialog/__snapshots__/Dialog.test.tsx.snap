// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog it should match snapshot 1`] = `
<DocumentFragment>
  <div
    style="position: fixed; top: 1px; left: 1px; width: 1px; height: 0px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; border-width: 0px; display: none;"
  />
</DocumentFragment>
`;
